doc/changelog/03-notations/11120-master+refactoring-application-printing.rst

```diff
@@ -0,0 +1,16 @@
+- **Fixed:** Inheritance of implicit arguments across notations made
+  uniform in parsing and printing. To the exception of notations of
+  the form ``Notation "..." := @foo`` and ``Notation id := @foo`` which
```

Comment:
You could consider using the Sphinx support for things like:

```suggestion
  the form ``Notation @string := @foo`` and ``Notation @ident__1 := @@ident__2`` which
```

Later on you can also use the `@args` token.

Reply:
Strictly speaking the second ident should be a qualid. I'll update.